static/js/stream_list.js

```diff
@@ -121,63 +103,40 @@ exports.build_stream_list = function () {
         return;
     }
 
-    streams = filter_streams_by_search(streams);
+    // The main logic to build the list is in stream_sort.js, and
+    // we get thre lists of streams (pinned/normal/dormant).
+    var stream_groups = stream_sort.sort_groups(get_search_term());
+
+    if (stream_groups.same_as_before) {
+        return;
+    }
```

Comment:
I'm a bit worried about the `same_as_before` logic here, given the call in `remove_sidebar_row` to `sort_groups`.  

Though I guess the idea is that in that case, we just did an (incremental) rerender?